### PR TITLE
Fix report export to Excel with aggregated bool columns

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [n]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [n]
| Automated tests included?              | [y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Report export to excel with aggregated bool columns transforms data to Yes/No.

#### Steps to reproduce issue:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create report with aggregated bool column e.g. Emails Sent with settings:
obraz
3. Save and check the values in View Report:
obraz
4. Export report to Excel and check the data:
obraz

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Repeat steps to reproduce
3. Data in XLS should be numerical

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11298"></a>

### CODE REVIEW

File: app/bundles/ReportBundle/Model/ExcelExporter.php [hunk 1]
```diff
@@ -24,11 +24,9 @@ public function __construct(FormatterHelper $formatterHelper)
     }
 
     /**
-     * @param string $name
-     *
      * @throws \Exception
      */
-    public function export(array $reportData, $name)
+    public function export(array $reportData, string $name, string $output = 'php://output')
```
Comment (escopecz) [inline app/bundles/ReportBundle/Model/ExcelExporter.php]: What was the reason to make this change? Adding types to existing params is considered a BC break. The 3rd param is used for test only. I'm not a fan of modifying production code for testing purposes but I also don't see a way around it in this case. This change makes this method test-able 👍

File: app/bundles/ReportBundle/Tests/Fixtures.php [hunk 2]
```diff
@@ -164,6 +164,22 @@ public static function getStringType()
         return 'string';
     }
 
+    /**
+     * @return string
+     */
+    public static function getIntegerType()
```
Comment (escopecz) [inline app/bundles/ReportBundle/Tests/Fixtures.php]:
```suggestion
    public static function getIntegerType(): string
```

File: app/bundles/ReportBundle/Tests/Fixtures.php [hunk 3]
```diff
@@ -164,6 +164,22 @@ public static function getStringType()
...
+    {
+        return 'int';
+    }
+
+    /**
+     * @return string
+     */
+    public static function getBooleanType()
```
Comment (escopecz) [inline app/bundles/ReportBundle/Tests/Fixtures.php]:
```suggestion
    public static function getBooleanType(): string
```

**Review by escopecz — COMMENTED** [review]
Yes, this works! Removing the code regarding the `$header` array makes sense as it is not used. It is dead code. 

The fix also worked. I was able to replicate the problem and confirmed the fix works as promised:
Screen Shot 2022-07-01 at 17 01 32

Please check my comments in the code.

**Comment by escopecz** [comment]
Oh and BTW I had to use slightly different report config as yours caused SQL errors for me because the columns in the SUM aggregations were not in the SELECT part. This is what I did to test this:
Screen Shot 2022-07-01 at 17 12 54

File: app/bundles/ReportBundle/Model/ExcelExporter.php (hunk 1 above)
Comment (patrykgruszka) [reply app/bundles/ReportBundle/Model/ExcelExporter.php]: PHPSTAN enforces types in this change. If there is a workaround we can remove the type from `$name` parameter. Unfortunately I haven't found any way to test this export without making the output a parameter, so this is only for testing purposes.

**Comment by patrykgruszka** [comment]
@escopecz I there is an issue with saving the calculated columns and it's not related with this PR. I've described this problem in #11311.

**Comment by RCheesley** [comment]
I am unable to test this PR even with the setting that are shared above - I get an SQL error :(

**Comment by patrykgruszka** [comment]
@RCheesley even with the workaround from #11311 (steps 6-9)?

**Review by RCheesley — DISMISSED** [review]
Thanks for the workaround re. the report fields - once I re-saved I was able to confirm the bug fix! 🚀

**Comment by RCheesley** [comment]
As this has two good tests and a code review, assuming that @escopecz is happy with [this reply](https://github.com/mautic/mautic/pull/11298#discussion_r912731492) then we are good to go, if not then it needs that issue addressed. I have left it as Code Review for this reason, but once that is addressed we can set it to RTC.

File: app/bundles/ReportBundle/Model/ExcelExporter.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/ReportBundle/Model/ExcelExporter.php]: If you'd add back the type declaration via comment `@param string $name` then you can remove the type from the `$name` param in the method signature. Does it make sense?

Comment (patrykgruszka) [reply app/bundles/ReportBundle/Model/ExcelExporter.php]: Yes, I've changed the type declaration. Should be good now.

**Comment by patrykgruszka** [comment]
@RCheesley I've made changes suggested by @escopecz. This PR should be ready to go now.

**Review by escopecz — APPROVED** [review]
Re-approving after the latest type fix commits 👍 Thanks!

**Pull request merged**